tasks/firewall/ufw_enable.yml

```diff
@@ -13,6 +13,10 @@
       state: enabled
       policy: allow
 
+  - name: reset ufw rules
+    community.general.ufw:
+      state: reset
+
```

Comment:
Quite brute, but it works :D

Reply:
From the man page of ufw:
> **reset**  Disables and resets firewall to installation defaults.

This will allow anything. I tested it. All policies are set to ACCEPT and all rules are set to "ACCEPT anywhere anywhere".
So this task could even replace the task above (policy allow). But we need an activate task afterwards. The reset task disables the whole firewall. Adding rules after does not activate the firewall again.

`ufw reset` also makes backups of the current rules. This could mess up /etc/ufw after a while.

How can i revert my approvement? :D

Reply:
I researched this quite extensively when I was working on #164 
There is no perfect way to do this, at least at the moment.

The suggested course of action is, as @crosscodr said, to move the `- name: Activate ufw` task at the bottom of the file, so that reset happens -> rules added -> firewall activated

I guess a different approach would be to parse the number of rules from `ufw status numbered` and then delete them all in a loop before adding them again, but I don't like this either because it will be displayed as changed state even when changing nothing.

Reply:
The point is if the default deny is set before you get locked out, that's why you have actually allow default, reset it and then turn it on again. I'll check out if it's possible to somehow fix the stuff that happens in /etc/ufw

Reply:
Is there any reason we should stick with UFW? I know it's more work, but I think creating a template for iptables and restoring it on the host, might be the way to go. We could use [this](https://docs.ansible.com/ansible/2.10/collections/community/general/iptables_state_module.html) and the template would include only the rules required by the enabled features, controlled by ansible variables (ex not including coturn rules if coturn not enabled). We could also use the [built-in iptables module](https://docs.ansible.com/ansible/2.10/collections/ansible/builtin/iptables_module.html) to directly manipulate the rules, but I prefer the template idea. I could help creating the template, if we decide migrating away from UFW is a good option.

Reply:
Yeah, I would also prefer direct use of iptables-templates :)

Reply:
I can write something up till the end of the week.

Reply:
We ran into some issues running bigbluebutton with iptables, but if you can make it work I am ok with that :)

Reply:
> 
> 
> I can write something up till the end of the week.

Feel free to ping me and let me know how I can assist.

Reply:
@d3hof ufw is just a wrapper around iptables, so if ufw works with bbb, iptables will do also :)
@ChrisKar96, thank you. I'll put up some WIP-PR soon.